Support for updating builder based on the images

pacostas opened the pull request:
<!-- Thanks for contributing. To speed up the process of reviewing your pull
request please provide us with the following information: -->

## Summary
<!-- A short explanation of the proposed change -->
Support of the RFC https://github.com/buildpacks/rfcs/blob/main/text/0128-multiarch-builders-and-package.md#package-a-multi-arch-builder that deprecates stacks and introduces build and run images on the builder.

## Use Cases
<!-- An explanation of the use cases your change enables -->
This PR:
* alters the update builder command in order to support the update of the run and build images
* It does not alter any of the previous behavior, therefore is backward comptible and supports stacks too or both (stacks and build/run images)

## Checklist
<!-- Please confirm the following -->
* [x] I have viewed, signed, and submitted the Contributor License Agreement.
* [ ] I have linked issue(s) that this PR should close using keywords or the Github UI (See [docs](https://docs.github.com/en/github/managing-your-work-on-github/linking-a-pull-request-to-an-issue))
* [x] I have added an integration test, if necessary.
* [x] I have reviewed the [styleguide](https://github.com/paketo-buildpacks/community/blob/main/STYLEGUIDE.md) for guidance on my code quality.
* [x] I'm happy with the commit history on this PR (I have rebased/squashed as needed).

jericop commented:
This looks good as is and we may want to merge and release it since it is a non-breaking change. However, it doesn't support updating multiple run images when not specified as stack mirror images. Not a big deal at present, but it should be addressed.

Another thing I noticed while reviewing this change is that we are using the semver tag from the build image when updating the run image instead of finding the latest sermver tag. This assumes that build and run images are always built and released at the same time. We can certainly keep that behavior if that is what is expected, but it may be possible to have run images in different registries with different semvers where this approach would not work. The above approach would not work in that scenario. Not sure it should block this PR from moving forward and it should probably be handled in a different PR to keeps things simple. Just wanted to call this out and see if there is historical context for the current approach. Any thoughts @robdimsdale

pacostas commented:
You are right it does not support multiple run images. I'll try to patch it and let you know :)

pacostas commented:
The changes since the last commit that add updating multiple images, are on this commit https://github.com/paketo-buildpacks/jam/pull/409/commits/405db805326bb846ef35fc8a6452e6310e14fb85
